src/main/java/com/google/sps/servlets/ReviewsServlet.java

```diff
@@ -0,0 +1,122 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import static com.google.sps.Constants.*;
+import com.google.sps.Review;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Servlet that handles customer reviews.
+ */
+@WebServlet("/reviews")
+public class ReviewsServlet extends HttpServlet {
+  Gson gson;
+  DatastoreService datastore;
+
+  public ReviewsServlet() throws IOException {
+    gson = new Gson();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  /** Get review from review form and store it in Datastore under its Business entity. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get review input from review form.
+    String userID = getParameter(request, "name", "anonymous"); // Accept input name for now, should be user Datastore key, to be implemented later
+    String comment = getParameter(request, "review", "404");
+    int rating = request.getParameter("rating");
+
+    // Get current timestamp for time when review is posted.
+    LocalDateTime dateTime = LocalDateTime.now();
+
+    // Get existing reviews key list.
+    List<Key> reviewsKeyList = getReviewsKeyList();
+
+    // Create review entity and get its key.
+    Entity reviewEntity = new Entity("Review");
+    reviewEntity.setProperty("userID", userID);
+    reviewEntity.setProperty("comment", comment);
+    reviewEntity.setProperty("rating", rating);
+    reviewEntity.setProperty("dateTime", dateTime);
+    datastore.put(reviewEntity);
+    Key reviewKey = reviewEntity.getKey();
+
+    // Append key to its Business entity.
+    reviewsKeyList.add(reviewKey);
+    businessEntity.setProperty(BUSINESS_REVIEWS, gson.toJson(reviewsKeyList));
```

Comment:
Have you initialised or fetched the Business entity from the datastore yet? 

`String businessID = request.getParameter("businessID"); //This could be something else, depending on what @Jacob uses as the ID
    long convertedID = Long.parseLong(businessID);
    Key businessKey = KeyFactory.createKey("Business", convertedID);
    Entity results = datastore.get(key);`

Reply:
The fetching of the business entity is done in the getReviewsKeyList() function. I'm not sure what the purpose of the code you posted is? Are you suggesting to parse the businessID as a long instead?

Reply:
I think this is fine